packages/starknet-snap/src/state/transaction-state-manager.ts

```diff
@@ -0,0 +1,305 @@
+import type { constants, TransactionType } from 'starknet';
+import {
+  TransactionFinalityStatus,
+  TransactionExecutionStatus,
+} from 'starknet';
+import { assert, enums, number } from 'superstruct';
+
+import type { Transaction, SnapState } from '../types/snapState';
+import { TransactionStatusType } from '../types/snapState';
+import type { IFilter } from './filter';
+import {
+  BigIntFilter,
+  ChainIdFilter as BaseChainIdFilter,
+  StringFllter,
+  Filter,
+} from './filter';
+import { StateManager, StateManagerError } from './state-manager';
+
+export type ITxFilter = IFilter<Transaction>;
+
+export class ChainIdFilter
+  extends BaseChainIdFilter<Transaction>
+  implements ITxFilter {}
+
+export class ContractAddressFilter
+  extends BigIntFilter<Transaction>
+  implements ITxFilter
+{
+  dataKey = 'contractAddress';
+}
+export class SenderAddressFilter
+  extends BigIntFilter<Transaction>
+  implements ITxFilter
+{
+  dataKey = 'senderAddress';
+}
+
+export class TxHashFilter
+  extends BigIntFilter<Transaction>
+  implements ITxFilter
+{
+  dataKey = 'txnHash';
+}
+
+export class TxTimestampFilter
+  extends Filter<number, Transaction>
+  implements ITxFilter
+{
+  _apply(data: Transaction): boolean {
+    // The timestamp from the data source is in seconds, but we are comparing it in milliseconds
+    return this.search !== undefined && data.timestamp * 1000 >= this.search;
```

Comment:
here we look for all transaction that happened after the search timestamp ? E.g. if I search for 1000 seconds in the past I put 1000 in timestamp when calling findTransactions ?

Reply:
the filter allow you to search the txn that larger than or equal to the given timestamp

e.g i wanna list txn from today till to 7 days ago